### PR TITLE
Cleanup project config after deleting .git directory

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfonov@redhat.com>

### What does this PR do?
After removing .git directory project doesn't support Git VCS, so we need to cleanup project configuration.
To keep consistency we should also delete project types related to the VCS (like `git` and `pullrequest` ), related attributes (branch name, remote origin and etc) and source configuration.

### What issues does this PR fix or reference?
#7562

### CODE REVIEW

**Comment by vparfonov** [comment]
ci-test

File: plugins/plugin-pullrequest-parent/che-plugin-pullrequest-server/src/main/java/org/eclipse/che/plugin/pullrequest/server/ContributionProjectTypeCleaner.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      projectManager.update(projectConfig);
+    } catch (NotFoundException e) {
+      // throw if given project not found
+      LOG.error(
+          "Project {} not found for removing {} type",
+          projectPath,
+          CONTRIBUTION_PROJECT_TYPE_ID,
+          e);
+    } catch (ServerException e) {
+      e.printStackTrace();
```
Comment (mmorhun) [inline plugins/plugin-pullrequest-parent/che-plugin-pullrequest-server/src/main/java/org/eclipse/che/plugin/pullrequest/server/ContributionProjectTypeCleaner.java]: Probably `LOG.error` with simple explanation in message would be better.

File: wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectTypeCleaner.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              null,
+              null);
+
+      projectManager.update(projectConfig);
+    } catch (NotFoundException e) {
+      // throw if given project not found
+      LOG.error(
+          "Project {} not found for removing {} type", projectPath, GitProjectType.DISPLAY_NAME, e);
+    } catch (ServerException e) {
+      e.printStackTrace();
```
Comment (mmorhun) [inline wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitProjectTypeCleaner.java]: The same as comment above about `LOG.error`

**Review by mmorhun — APPROVED** [review]

File: plugins/plugin-pullrequest-parent/che-plugin-pullrequest-server/src/main/java/org/eclipse/che/plugin/pullrequest/server/ContributionProjectTypeCleaner.java (hunk 1 above)
Comment (vparfonov) [reply plugins/plugin-pullrequest-parent/che-plugin-pullrequest-server/src/main/java/org/eclipse/che/plugin/pullrequest/server/ContributionProjectTypeCleaner.java]: thx, i will fix

**Comment by vparfonov** [comment]
ci-test

**Comment by vparfonov** [comment]
ci-test

**Comment by vparfonov** [comment]
ci-test

**Review by dkulieshov — APPROVED** [review]

**Comment by vparfonov** [comment]
ci-test

**Review by musienko-maxim — APPROVED** [review]

**Pull request merged**